File: linked-list/reverselinklistt.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev=NULL;
        ListNode* curr=head;
        ListNode* next_next=NULL;
        while(curr!=NULL){
            next_next=curr->next;
            curr->next=prev;
            prev=curr;
            curr=next_next;

        }
        head=prev;
        return head;
    }
};

File: linked-list/reverselinklistt.js
class node{
    constructor(val,next){
        this.data=val;
        this.next=null;
    }

    let head= new node(val,next);

var reverseList = function(head) { //head is the first node.
    let current=head;
    let prev= null;
    let next =null;
    while(current!=null){
            next=current.next;
            current.next=prev;
            prev=current;
            current=next;
    }
    head=prev; //at last the prev variable containg the node is now contained by head variable. so that we can call the linklist by the head.
    return head;
};
